wavesexchange_topic/src/lib.rs

```diff
@@ -303,10 +321,91 @@ mod parse_and_format {
                             return Err(TopicParseError::InvalidLeasingBalanceTopic);
                         }
                     }
+                    TopicKind::Pairs => {
+                        Topic::extract_exchange_pairs_from_query(&url)?;
+                    }
                 }
 
                 Ok(())
             }
+
+            fn extract_pairs_from_get_params(
+                pairs_in_get_params: Result<Vec<String>, ()>,
+            ) -> Result<ExchangePairs, TopicParseError> {
+                // topic://pairs?pairs[]=amount_asset_id/price_asset_id&pairs[]=amount_asset_id1/price_asset_id1
+
+                let mut pairs = vec![];
+
+                match pairs_in_get_params {
+                    Err(_) => return Err(TopicParseError::InvalidExchangePairs),
+                    Ok(q_pair) => {
+                        for p in q_pair {
+                            let pair = p.split("/").collect::<Vec<&str>>();
+
+                            if pair.len() != 2 {
+                                return Err(TopicParseError::InvalidExchangePairs);
+                            }
+
+                            let mut pair = pair.iter();
+
+                            if let (Some(amount_asset), Some(price_asset)) =
+                                (pair.next(), pair.next())
+                            {
+                                let p = ExchangePair {
+                                    amount_asset: (*amount_asset).into(),
+                                    price_asset: (*price_asset).into(),
+                                };
+                                pairs.push(p);
+                            }
+                        }
+                    }
+                }
+
+                Ok(ExchangePairs { pairs })
+            }
+
+            pub fn extract_exchange_pairs_from_query(
```

Comment:
Why is this method public? It is not a part of Topic's public API. Either make private, or at most pub(crate).

Reply:
done